org.mwc.debrief.legacy/src/Debrief/ReaderWriter/Replay/ImportReplay.java

```diff
@@ -785,6 +791,20 @@ public HiResDate readLine(final String theLine) throws java.io.IOException
 		{
 			res = processReplayFix((ReplayFix) thisObject);
 		}
+		else if (thisObject instanceof DynamicShapeWrapper)
+		{
+			proccessShapeWrapper(thisOne, thisObject);
+			DynamicShapeWrapper thisWrapper = (DynamicShapeWrapper) thisObject;
+			String trackName = thisWrapper.getTrackName();
+			DynamicShapeLayer dsl = (DynamicShapeLayer) getLayerFor(trackName);
+			if (dsl == null) 
+			{
+				dsl = new DynamicShapeLayer();
+				dsl.setName(trackName);
+			}
+			addToLayer(thisWrapper, dsl);
+			addLayer(dsl);

```

Comment:
hmm, if an existing layer has been found (dsl != null), won't line 806 make it get added twice?

I would expect addLayer(dsl) to only  occur inside the if-block.